kale/interpret/uncertainty_quantiles.py

```diff
@@ -0,0 +1,1931 @@
+        font_size_2,
+        bin_label_locs,
+        x_axis_labels,
+        num_bins,
+        uncertainty_types_list,
+        all_sample_percs,
+        all_sample_label_x_locs,
+        max_bin_height,
+    )
+
+
+def format_plot(
+    ax,
+    save_path: Optional[str],
+    show_sample_info: str,
+    to_log: bool,
+    circ_patches: List[patches.Patch],
+    y_lim: float,
+    y_lim_min: float,
+    turn_to_percent: bool,
+    x_label: str,
+    y_label: str,
+    font_size_1: int,
+    font_size_2: int,
+    bin_label_locs: List[float],
+    x_axis_labels: List[str],
+    num_bins: int,
+    uncertainty_types_list: List[List[str]],
+    all_sample_percs: List[List[float]],
+    all_sample_label_x_locs: List[List[Any]],
+    max_bin_height: float,
+    comparing_q: bool = False,
+) -> None:
+    """
+    This function takes a matplotlib Axes object and formats the plot according to the provided parameters.
+
+    Args:
+        ax: A matplotlib axes object to be formatted.
+        save_path: The path where the plot should be saved. If None, the plot will be shown using plt.show().
+        show_sample_info: Determines how sample information is displayed. Can be "None", "Average", or "All".
+        to_log: If True, sets the y-axis to log scale.
+        circ_patches: List of matplotlib patches to be added to the legend.
+        y_lim: The upper limit for the y-axis.
+        y_lim_min: The lower limit for the y-axis.
+        turn_to_percent: If True, converts y-axis values to percentages.
+        x_label: The label for the x-axis.
+        y_label: The label for the y-axis.
+        font_size_1: The font size for the axis labels.
+        font_size_2: The font size for the tick labels.
+        bin_label_locs: The x-axis locations of the bin labels.
+        x_axis_labels: The labels for the x-axis.
+        num_bins: The number of bins.
+        uncertainty_types_list: The list of uncertainty types.
+        all_sample_percs: The percentage of samples for each bin.
+        all_sample_label_x_locs: The x-axis locations of the sample percentage labels.
+        max_bin_height: The maximum height of a bin in the plot.
+        comparing_q: If True, it uses a ticker.FixedFormatter for the x-axis.
+
+    Returns:
+        None
+    """
+
+    # Show the average samples on top of boxplots, aligned. if lots of bins we can lower the height.
+    if show_sample_info != "None":
+        for idx_text, perc_info in enumerate(all_sample_percs):
+            if show_sample_info == "Average":
+                ax.text(
+                    all_sample_label_x_locs[idx_text],
+                    max_bin_height * 0.8,  # Position
+                    r"$\bf{PSB}$" + ": \n" + r"${} \pm$".format(perc_info[0]) + "\n" + r"${}$".format(perc_info[1]),
+                    verticalalignment="bottom",  # Centered bottom with line
+                    horizontalalignment="center",  # Centered with horizontal line
+                    fontsize=25,
```

Comment:
Hard coded. Check what other libraries do with this. If not time to fix it, log it in the discussion.

Reply:
added to discussion